cli/internal/kubernetes/upgrade.go

```diff
@@ -85,10 +96,23 @@ type Upgrader struct {
 	outWriter        io.Writer
 	tfUpgrader       *upgrade.TerraformUpgrader
 	log              debugLog
+	upgradeID        string
 }
 
 // NewUpgrader returns a new Upgrader.
-func NewUpgrader(ctx context.Context, outWriter io.Writer, log debugLog) (*Upgrader, error) {
+func NewUpgrader(ctx context.Context, outWriter io.Writer, log debugLog, upgradeCmdKind UpgradeCmdKind) (*Upgrader, error) {
+	upgradeID := "upgrade-" + time.Now().Format("20060102150405") + "-" + strings.Split(uuid.New().String(), "-")[0]
+	if upgradeCmdKind == UpgradeCmdKindCheck {
+		upgradeID += "-check"
```

Comment:
recap from our short conversation online: 
- change to "-tmp"
- add comment why we need to handle those folder separately.